shared/domains/chelonia/utils.js

```diff
@@ -731,6 +743,14 @@ export function eventsAfter (contractID: string, sinceHeight: number, limit?: nu
             }
             break
           }
+          case 'eod': {
+            if (remainingEvents === 0 || sinceHeight >= latestHeight) {
+              controller.close()
+            } else {
+              controller.error(new Error('Unexpected end of data'))
+            }
```

Comment:
Similar question, is a `throw` needed here? If not, why not? Could a comment be added at the first instance of this pattern that explains why the error isn't thrown?